### DISCUSSION TITLE
link already recorded timesheet to a (later created) task

### DISCUSSION THREAD

**Discussion by buzter** [kimai/Paid-plugins#7] [Questions & Answers]
Hello,

we will often have the following case:

We receive an unexpected phone call and start the time recording without knowing what to expect. The call "creates" a new task for an already existing project. After the phone call we stop the time recording. Only after that we get around to assign the freshly "created" task to the existing project.

The already recorded timesheet (initial call described above) I would like to link to the just created task. Is this possible?

I read the documentation and tried in our kimai installation - whitout success.

Until now only a manual change of the database could help. I assigned the before recorded timesheet to the later created task there. It seems to work properly but I would like to avoid manual changes of the database so no inconsistencies arise. Nonetheless, this would be a one-off solution. But I assume we will have the case described above very often.

Best regards

**Comment by pbksol** [comment]
A task can create a timesheet. A timesheet cannot create a task. That's how the current workflow is designed in Kimai, you're trying to use it in a way it's just not designed for.
I sounds a bit unusual to me to bill time on a not existing task, tbh.
Shouldn't the initial call be billed to something like "general", "call support" which is way more specific to what the tech actually did? Or maybe "taking customer request by phone". Then the task is created and the one actually working on this task can then generate billed time for that.

**Comment by buzter** [comment]
> A task can create a timesheet. A timesheet cannot create a task. That's how the current workflow is designed in Kimai, you're trying to use it in a way it's just not designed for.

I understand that without ifs and buts.

My issue is not to create a task by a timesheet. I want to assign an existing timesheet to a (e.g. later created) task. By the way the task doesn't have to be created afterwards at all. The call from my example above can refer to an existing task of a project. But unfortunately the team member has created the timesheet without associating it with the task. So we have to do it later when the "mistake" comes to light.

**Comment by kevinpapst** [comment]
You are both right, the example might habe been a bit misleading.

I think it is mostly a matter of getting used to the workflow (or understand the limitations of the plugin).
But I also understand why someone wants to manually assign timesheets to tasks... even though all methods (restart, copy, start from task, start from dashboard) already link the new timesheet to the task.

I am fine with the feature request, that's why it already exists:
https://github.com/kevinpapst/kimai-plugins/issues/10

**Reply by buzter** [reply]
I had already seen this. To be on the safe side, I just wanted to make sure during the discussions that I hadn't overlooked anything. Many Thanks

**Reply by william-letton** [reply]
The link above gives a 404 error, so I couldn't follow it. From this thread and testing the desktop version of the software, it looks like:

1) Associating an activity with a task will not automatically associate any time spent on that activity to that task.
2) Instead the task itself has to be started, live, under Tasks.
3) Starting the task under Tasks will create an activity, which can then later be restarted under 'My times' and will continue to be associated with that task.
4) There is no visual difference under 'My times' between an activity record that is associated with the task and one that isn't.
5) There is no way to associate a completed activity with a task.

Is this correct? If so this is unfortunate, as we log our activities after they are completed rather than starting and stopping them in the application. This would render the Task tracking functionality of no use to us.
